### PR TITLE
Wizard apprentices made their own thing

### LINKED ISSUE #10801: Wizard Apprentices don't get wiz hud elements
(WEB REPORT BY: gutlesscoward REMOTE: 172.93.109.202:7777)

> Revision 9709e5d58d69e0e097496e3af7870ca97eb44066
> 
> General description of the issue
> Apprentices made using the apprentice scroll can't see the little wizard hat icon that denotes someone as a wizard. this leads to problems when one is trying to stealth and the other is murdering everything on sight.
> 
> What you expected to happen
> I expected my apprentice to know that I was a wizard despite the disguise
> 
> What actually happened
> I got murdered to death by my apprentice, who didn't know I was a wizard.
> 
> Steps to reproduce if possible
> 1. Be a wizard
> 2. Buy apprentice scrolls. I picked muscle wizard.
> 3. remove your robes and wear a disguise.
> 4. meet up with your apprentice

### PR DESCRIPTION
**Pull request by Sprok0**
Wizard apprentices are now their own antagonist type, rather than traitors with spells. They have a little red hat icon which can be seen by wizards, can see the wizard blue hat icon, and have their own section in the round end scoreboard below the wizards. They can also ride broomsticks like real wizards!

Fixes #10801

### CODE REVIEW

**Comment by Shadowmech88** [comment]
Can apprentices see the hat icons of other apprentices?

**Comment by Sprok0** [comment]
Apprentices can see apprentice and wizard icons, wizards can see wizard and apprentice icons

**Comment by SonixApache** [comment]
Can an apprentice see any wizard's icon, or only the wizard they are apprenticed to? And vice versa, really.

**Comment by Sprok0** [comment]
Right now, all apprentices see all wizard icons, all wizards see all apprentice icons

**Comment by ghost** [comment]
AAAAAAAAA this was much better implemented in antag HUD datums

**Comment by ghost** [comment]
Well, the HUD, anyway, the rest of this seems fine

**Comment by Sprok0** [comment]
All I did was copy the hud code from rev and changed variable names, feel free to rip that out when your thing works

File: code/modules/admin/admin.dm [hunk 1]
```diff
@@ -1176,6 +1176,10 @@ var/global/floorIsLava = 0
 		if (ticker.mode.config_tag == "wizard")
 			return 2
 		return 1
+	if(isapprentice(M))
```
Comment (ComicIronic) [inline code/modules/admin/admin.dm]: Merge this into the above if()

File: code/game/gamemodes/wizard/wizard.dm [hunk 2]
```diff
@@ -378,9 +439,24 @@ Made a proc so this is not repeated 14 (or more) times.*/
 						if(I.icon_state == "wizard")
 							wizard_mind.current.client.images -= I
 
+		for(var/datum/mind/apprentice_mind in apprentices)
+			if(apprentice_mind.current)
+				if(apprentice_mind.current.client)
+					for(var/image/I in apprentice_mind.current.client.images)
+						if(I.icon_state == "wizard" || I.icon_state == "apprentice")
```
Comment (ComicIronic) [inline code/game/gamemodes/wizard/wizard.dm:446]: This is a really poor way of doing it. Not only are you introducing a bug if the icon states are ever changed in the file, but it's just lazy. Use a list of images to keep track of the wizard icon ones.

Comment (Sprok0) [reply code/game/gamemodes/wizard/wizard.dm:446]: Honestly I have no idea how this works, it's just rev code with the names changed

**Comment by clusterfack** [comment]
Kill gamemodes and start anew

**Pull request merged**